### PR TITLE
Feature/add use react router breadcrumbs

### PR DESCRIPTION
**Pull request by PanopticaRising**

### CODE REVIEW

File: src/Contexts/BreadcrumbContext.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    return (
+        <BreadcrumbContext.Provider value={{
+            breadcrumbLookup, updateBreadcrumbLookup: updateBreadcrumbLookupWrapper
+        }}>
+            {children}
+        </BreadcrumbContext.Provider>
+    );
+};
+
+export const useCurrentProblemState = () => React.useContext(BreadcrumbContext);
```
Comment (red-tt-bear) [inline src/Contexts/BreadcrumbContext.tsx:39]: useCurrentProblemState? is this copy pasta?

File: src/Contexts/BreadcrumbContext.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+export const useBreadcrumbLookupContext = () => React.useContext(BreadcrumbContext);
+
+export const BreadcrumbLookupProvider: React.FC<Props>  = ({ children }) => {
+    const [breadcrumbLookup, updateBreadcrumbLookup] = useState<BreadcrumbState>({});
+
+    const updateBreadcrumbLookupWrapper = (state: BreadcrumbState) => updateBreadcrumbLookup(oldstate => ({...oldstate, ...state}));
+
+    return (
+        <BreadcrumbContext.Provider value={{
+            breadcrumbLookup, updateBreadcrumbLookup: updateBreadcrumbLookupWrapper
```
Comment (red-tt-bear) [inline src/Contexts/BreadcrumbContext.tsx:32]: Thanks for switching to update!

File: src/Contexts/BreadcrumbContext.tsx (hunk 1 above)
Comment (PanopticaRising) [reply src/Contexts/BreadcrumbContext.tsx:39]: It obviously didn't work and I readded it on line 17 :joy:

File: src/Courses/TopicSettings/TopicSettingsPage.tsx [hunk 3]
```diff
@@ -93,6 +96,8 @@ export const TopicSettingsPage: React.FC<TopicSettingsPageProps> = ({topic: topi
             newTopic.questions.push(newProb);
 
             setTopic(newTopic);
+            // Name should enver be updated here, so no need to cache.
```
Comment (red-tt-bear) [inline src/Courses/TopicSettings/TopicSettingsPage.tsx:99]: never*

Comment (red-tt-bear) [reply src/Courses/TopicSettings/TopicSettingsPage.tsx:99]: Thanks for the comment!

File: src/Courses/TopicSettings/TopicSettingsPage.tsx [hunk 4]
```diff
@@ -151,6 +156,8 @@ export const TopicSettingsPage: React.FC<TopicSettingsPageProps> = ({topic: topi
 
             setTopic(newTopic);
             setSelected(selected => selected instanceof ProblemObject ? new ProblemObject({...selected}) : selected);
+            // Name should enver be updated here, so no need to cache.
```
Comment (red-tt-bear) [inline src/Courses/TopicSettings/TopicSettingsPage.tsx:159]: Copy pasta!

File: src/Courses/TopicSettings/TopicSettingsPage.tsx (hunk 3 above)
Comment (PanopticaRising) [reply src/Courses/TopicSettings/TopicSettingsPage.tsx:99]: // ashamed.jpg

**Review by red-tt-bear — APPROVED** [review]

**Pull request merged**